test/validation/api/align/align.c

```diff
@@ -0,0 +1,131 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2024 Nokia
+ */
+
+#include <odp_api.h>
+#include <odp_cunit_common.h>
+
+#include <stddef.h>
+#include <stdint.h>
+
+typedef struct ODP_PACKED {
+	uint8_t a;
+	uint8_t b;
+	uint16_t c;
+	uint32_t d;
+} test_type_t;
```

Comment:
In addition to this struct, there could be a struct with "holes" in it:

typedef struct ODP_PACKED {
	uint8_t a;
	uint16_t b;
	uint8_t c;
	uint32_t d;
} test_type_2_t;